client/client_test.go

```diff
@@ -1346,3 +1349,72 @@ func tmpdir(appliers ...fstest.Applier) (string, error) {
 	}
 	return tmpdir, nil
 }
+
+func testParallelBuilds(t *testing.T, sb integration.Sandbox) {
+	t.Parallel()
+	ctx, cancel := context.WithCancel(context.TODO())
+	defer cancel()
+
+	c, err := New(ctx, sb.Address())
+	require.NoError(t, err)
+	defer c.Close()
+
+	eg, ctx := errgroup.WithContext(ctx)
+
+	for i := 0; i < 3; i++ {
+		i := i
+
+		eg.Go(func() error {
+			srcDir, err := ioutil.TempDir("", "buildkit")
+			require.NoError(t, err)
+			defer os.RemoveAll(srcDir)
+
+			fn := fmt.Sprintf("test%d.txt", i)
+			exp := fmt.Sprintf("This is iteration %d\n", i)
+			err = ioutil.WriteFile(filepath.Join(srcDir, fn), []byte(exp), 0666)
+			require.NoError(t, err)
+
+			src := llb.Local("source")
+
+			run := llb.Image("busybox:latest").Dir("/src").Run(
+				llb.ReadonlyRootFS(),
+				// Everything goes to stderr for coherent/ordered logging.
+				// The `ls` records inodes in order to more easily spot clashes.
+				llb.Args([]string{"/bin/sh", "-c", "set -x ; 1>&2 echo Start " + fn + " 1>&2; ls -ilRt /src /out 1>&2 ; cat " + fn + " 1>&2 ; cat test*.txt 1>&2 ; stat test*.txt 1>&2 ; /bin/cp -v  " + fn + " /out/test.txt 1>&2 ; echo End " + fn + " 1>&2"}),
```

Comment:
`"-ec"`?

Reply:
I deliberately left out `-e` so that all the various probing and inspecting didn't cause the whole thing to fail too early to get the interesting information.